### DISCUSSION TITLE
Not getting GenerateReleasenotes testconsole to work - UnhandledPromiseRejectionWarning

### DISCUSSION THREAD

**Discussion by holtsmeister** [rfennell/AzurePipelines#1102] [Q&A] [answered]
I'm trying to launch the testconsole, as seen at 41:06 here: https://www.youtube.com/watch?v=xaV3dFoQdV8&ab_channel=DeveloperDays

I tried creating a new PAT token on my Azure account with full access, and wrote the following command (dummy pat):

`node GenerateReleaseNotesConsoleTester.js --filename build-settings.json --pat uewkwci6l32pifx573jrqlvdwti2s3v4ajnv6skqhq`

Getting this log with an error. Am I using the pat token in the wrong way? You didn't elaborate much on it in the video.

##vso[task.debug]agent.TempDirectory=undefined
##vso[task.debug]agent.workFolder=undefined
##vso[task.debug]loading inputs and endpoints
##vso[task.debug]loaded 0
##vso[task.debug]Agent.ProxyUrl=undefined
##vso[task.debug]Agent.CAInfo=undefined
##vso[task.debug]Agent.ClientCert=undefined
##vso[task.debug]Agent.SkipCertValidation=undefined
Starting Tag XplatGenerateReleaseNotes Local Tester
Command Line Arguements:
  --filename: build-settings.json
  --pat: ************************************************kqhq
  --githubpat:
  --bitbucketuser:
  --bitbucketsecret:
Loading settings from build-settings.json
Creating Azure DevOps API connections for https://dev.azure.com/richardfennell with 'allowRetries' set to 'true' and 'maxRetries' count to '3'
Creating the credential handler using override PAT (suitable for local testing or if the OAUTH token cannot be used)
(node:22772) UnhandledPromiseRejectionWarning: Error: Failed request: (401)
    at RestClient.<anonymous> (C:\Gitkod\AzurePipelines\Extensions\XplatGenerateReleaseNotes\V3\node_modules\typed-rest-client\RestClient.js:202:31)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Gitkod\AzurePipelines\Extensions\XplatGenerateReleaseNotes\V3\node_modules\typed-rest-client\RestClient.js:6:58)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:22772) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:22772) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by rfennell** [comment, marked as answer]
That command line looks fine. The issue is not the command line but the configuration file.

I can see from the log you are trying to connect to my test Azure DevOps instance, which I am sure is not what you are trying to do

> Creating Azure DevOps API connections for https://dev.azure.com/richardfennell with 'allowRetries' set to 'true' and 'maxRetries' count to '3'

You need to look at the contents of `build-settings.json`. It holds all the configuration information that Azure DevOps would normally provide e.g. environment variables such as the service URL, team project, build ids etc and task parameters.

**Reply by holtsmeister** [reply]
Time spent on the readme = 0 sec, sorry about that.

Worth noting is that the readme says:

> In the settings file make sure the `TeamFoundationCollectionUri`, `TeamProject` and `BuildID` are set to the build you wish to run against, and that the `ReleaseID` is empty.

But nothing about the other fields. releaseDefinitionId was prepopulated (I just removed that number) Also, I had to put something in "overrideStageName", otherwise I got "stage undefined"-ish errors. Doesn't matter what I put there for it to work.

Still have some exploring to do, I might realize why all this is. But as a noobie I'd like to be even more hand-held in a readme.

**Reply by rfennell** [reply]
Yes, a refresh of that readme would not go amiss, there are a number of new parameters.

Glad you got going though
